packages/datadog-instrumentations/src/fastify.js

```diff
@@ -27,6 +31,11 @@ function wrapFastify (fastify, hasParsingEvents) {
       app.addHook('preHandler', preValidation)
     }
 
+    // Hack to check if the Code Origin for Spans feature is enabled
+    if (codeOriginEnabledChannel.hasSubscribers) {
```

Comment:
we try to avoid importing `dd-trace` code in instrumentations unless it's absolutely necessary. I'm referring to `const { entryTag } = require('../../dd-trace/src/code_origin')`. 

You might be able to achieve the same with something like
```suggestion
   app.addHook('onRoute', function (routeOptions) {
      if (codeOriginEnabledChannel.hasSubscribers) {
         codeOriginEnabledChannel.publish(routeOptions)
      }
   })
```

and change the code in `packages/datadog-plugin-fastify/src/index.js` to:

```javascript
this.addSub('datadog:code-origin-enabled', routeOptions => {
  if (this._tracerConfig.codeOriginForSpansEnabled) {
    if (!routeOptions.config) routeOptions.config = {}
    routeOptions.config[kCodeOriginForSpansTagsSym] = entryTag(onRoute)
  }
})
```

Reply:
Fixed 👍